docs/install-midstream.md

```diff
@@ -0,0 +1,151 @@
+
+```yaml
+# CatalogSource for 1.24
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  # Replace the version with whatever version you want to install
+  name: serverless-operator-v1-24-0
+  namespace: openshift-marketplace
+spec:
+  displayName: Serverless Operator
+  # Replace the version with whatever version you want to install
+  image: registry.ci.openshift.org/knative/openshift-serverless-v1.24.0:serverless-index
+  publisher: Red Hat
+  sourceType: grpc
+---
+# CatalogSource for 1.25
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  # Replace the version with whatever version you want to install
+  name: serverless-operator-v1-25-0
+  namespace: openshift-marketplace
+spec:
+  displayName: Serverless Operator
+  # Replace the version with whatever version you want to install
+  image: registry.ci.openshift.org/knative/openshift-serverless-v1.25.0:serverless-index
+  publisher: Red Hat
+  sourceType: grpc
+---
+# CatalogSource for main
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  # Replace the version with whatever version you want to install
+  name: serverless-operator-nightly
+  namespace: openshift-marketplace
+spec:
+  displayName: Serverless Operator
+  # Replace the version with whatever version you want to install
+  image: registry.ci.openshift.org/knative/openshift-serverless-nightly:serverless-index
+  publisher: Red Hat
+  sourceType: grpc
+```
+
+__CatalogSource images aren't bound to a specific `patch` version, every commit to a specific branch will actually
+change the `vx.y.0` image (they are streams of major versions).__
+
+### Installation
+
+Apply the following subscription referencing the specific catalog source you want to install:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: serverless-operator
+  namespace: openshift-serverless
+spec:
+  channel: stable
+  name: serverless-operator
+  # Change the CatalogSource name
+  source: serverless-operator-v1-24-0
+  sourceNamespace: openshift-marketplace
+  # Automatic is fine and safe since there is only one version in the referenced catalog
+  installPlanApproval: Automatic
+```
+
+### Upgrade
+
+Apply the following subscription referencing the specific catalog source you want to upgrade to:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: serverless-operator
+  namespace: openshift-serverless
+spec:
+  channel: stable
+  name: serverless-operator
+  # Change the CatalogSource name
+  source: serverless-operator-v1-25-0
+  sourceNamespace: openshift-marketplace
+  # Automatic is fine and safe since there is only one version in the referenced catalog
+  installPlanApproval: Automatic
+```
+
+### Downgrade
+
+OLM doesn't support downgrades, so referencing in a Subscription a previous version isn't supported, so we need a few
```

Comment:
I'd write does not in bold :smile: (**does not**)

Reply:
I am also not sure if we should really document this, if the underlying framework has it.

Perhaps we can move this to a different markdown-file, and decorate it with warnings.

I am a bit afraid of folks reading and trying this, while we (as a product) are not really supporting that.


It is for sure valuable information, but perhaps make some more noise on what we want and what not

Reply:
We're testing this downgrade path and it is the only viable option we're recommending

Reply:
See SRVCOM-2037 and our downgrade tests

Reply:
I am ok adding this here. Specificially this is our own midstream repo. No product doc.

I am just concerned of a boomerang. Generally good to add some extra "warning" to path like this.